crates/provider-blobstore-nats/src/config.rs

```diff
@@ -0,0 +1,449 @@
+    #[serde(default)]
+    pub storage_config: Option<StorageConfig>,
+
+    /// Write operation timeout configuration
+    #[serde(default)]
+    pub max_write_wait: Option<u64>,
+}
+
+/// NATS Object-store is backed by its Key-Value store; so an object-store
+/// is identified by its backing bucket name.
+///
+/// Note that when storage config is provided via link configuration
+/// the following keys are expected:
+/// - `max_age` (optional): the maximum age of any object in the container, expressed in seconds; defaults to 10 years
+/// - `storage_type` (optional): the type of storage backend, File (default) and Memory
+/// - `num_replicas` (optional): how many replicas to keep for each object in a cluster, maximum 5; defaults to 1
+/// - `compression` (optional): whether the underlying stream should be compressed; defaults to false
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct StorageConfig {
+    /// Maximum age of any object in the container, expressed in nanoseconds
+    #[serde(default)]
+    pub max_age: Option<core::time::Duration>,
+    /// Maximum size of the object store container, expressed in bytes
+    #[serde(default)]
+    pub max_bytes: Option<i64>,
+    /// The type of storage backend, File (default) and Memory
+    #[serde(default)]
+    pub storage_type: Option<StorageType>,
+    /// How many replicas to keep for each object in a cluster, maximum 5
+    #[serde(default)]
+    pub num_replicas: Option<usize>,
+    /// Whether the underlying stream should be compressed
+    #[serde(default)]
+    pub compression: Option<bool>,
+}
+
+use std::str::FromStr;
+
+/// StorageType represents the type of storage backend to use; it maps the configuration value to the proper 'nats_async' type
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct StorageType(pub async_nats::jetstream::stream::StorageType);
+
+/// Implementing FromStr for StorageType to allow custom parsing from a string
+impl FromStr for StorageType {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "file" => Ok(StorageType(
+                async_nats::jetstream::stream::StorageType::File,
+            )),
+            "0" => Ok(StorageType(
+                async_nats::jetstream::stream::StorageType::File,
+            )),
+            "memory" => Ok(StorageType(
+                async_nats::jetstream::stream::StorageType::Memory,
+            )),
+            "1" => Ok(StorageType(
+                async_nats::jetstream::stream::StorageType::Memory,
+            )),
+            _ => Err(()),
+        }
+    }
+}
+
+/// Default implementation for [`StorageConfig`]
+impl Default for StorageConfig {
+    fn default() -> StorageConfig {
+        StorageConfig {
+            max_age: Some(core::time::Duration::from_secs(315_569_520)), // 10 years
+            max_bytes: Some(1_000_000_000),                              // 1 GB
```

Comment:
Were these values that you found somewhere, or just reasonable defaults? I could see the argument here to set the max_bytes to `-1` given that's what NATS does (unlimited)

Reply:
They had remained from the early versions; now they're set to -1